app/components/GeoLocationInput/GeoLocationModal/GeoLocationMapInput/index.tsx

```diff
@@ -0,0 +1,250 @@
+
+import styles from './styles.css';
+
+function geoAreaGroupKeySelector(geoArea: GeoArea) {
+    return geoArea.adminLevelTitle;
+}
+
+function geoAreaKeySelector(geoArea: GeoArea) {
+    return geoArea.id;
+}
+
+const GEOAREAS = gql`
+    query ProjectGeoAreas(
+        $projectId: ID!,
+        $ids: [ID!],
+    ) {
+        project(id: $projectId) {
+            geoAreas(ids: $ids) {
+                results {
+                    adminLevelTitle
+                    id
+                    regionTitle
+                    title
+                }
+            }
+        }
+    }
+`;
+
+const PROJECT_REGIONS = gql`
+    query ProjectRegions($projectId: ID!) {
+        project(id: $projectId) {
+            regions {
+                title
+                id
+                adminLevels {
+                    id
+                    level
+                    title
+                    geojsonFile {
+                        name
+                        url
+                    }
+                    boundsFile {
+                        name
+                        url
+                    }
+                }
+            }
+        }
+    }
+`;
+
+type ProjectRegion = NonNullable<NonNullable<NonNullable<ProjectRegionsQuery>['project']>['regions']>[number];
+export type AdminLevel = NonNullable<ProjectRegion['adminLevels']>[number];
+
+function regionKeySelector(d: ProjectRegion) {
+    return d.id;
+}
+
+function regionLabelSelector(d: ProjectRegion) {
+    return d.title;
+}
+
+interface Props {
+    className?: string;
+    projectId: string;
+    tempGeoAreas?: string[] | undefined;
+    onChange: (value: string[] | undefined) => void,
+    geoAreaOptions: GeoArea[] | null | undefined;
+    onGeoAreaOptionsChange: React.Dispatch<React.SetStateAction<GeoArea[] | null | undefined>>;
+}
+
+function GeoLocationMapInput(props: Props) {
+    const {
+        projectId,
+        className,
+        tempGeoAreas,
+        geoAreaOptions,
+        onGeoAreaOptionsChange,
+        onChange,
+    } = props;
+
+    const [selectedRegion, setSelectedRegion] = useState<string>();
+    const [selectedAdminLevel, setSelectedAdminLevel] = useState<AdminLevel>();
+
+    const [
+        getGeoAreas,
+    ] = useLazyQuery<ProjectGeoAreasQuery, ProjectGeoAreasQueryVariables>(
+        GEOAREAS,
+        {
+            onCompleted: (data) => {
+                onGeoAreaOptionsChange(data.project?.geoAreas?.results);
+            },
+        },
+    );
+
+    const variables = useMemo(() => ({
+        projectId,
+    }
+    ), [projectId]);
+
+    const {
+        data: projectRegions,
+        loading: projectRegionsPending,
+    } = useQuery<ProjectRegionsQuery, ProjectRegionsQueryVariables>(
+        PROJECT_REGIONS,
+        {
+            skip: !projectId,
+            variables,
+            onCompleted: (data) => {
+                const [topRegion] = data.project?.regions ?? [];
+                const [topAdminLevel] = topRegion.adminLevels ?? [];
```

Comment:
What if `topRegion` is undefined?

Reply:
if we don't have topRegion no region will be selected.